support/ebpf/dotnet_tracer.ebpf.c

```diff
@@ -171,6 +171,9 @@ ErrorCode unwind_one_dotnet_frame(PerCPURecord *record, DotnetProcInfo *vi, bool
   u64 code_header_ptr = pc;
 
   state->return_address = true;
+#if defined(__aarch64__)
+  state->can_use_lr = false;
+#endif
```

Comment:
This `#if` conditional code in various places is a bit clunky. Soince `return_address` and `can_use_lr` go hand in hand always except one location, would it make sense to add a helper. Something like `state_set_return_address` which sets both? This would also help to make sure we don't miss one or the other assignment in future.

Or perhaps even, `state_unwind_frame` which gets `fp`, `pc`, etc. But that might be worth a separate PR.

Reply:
What would you think about naming the helper macro something like `MARK_NONLEAF_FRAME` which just sets `return_address = true` and `lr_valid = false` ?

Reply:
I generally prefer `static inline` function in the header rather than a macro (better type checking, completion and compiler diagnostics). No strong opinion on the naming. Looking at `tracemgmt.h` there are some functions working on the same context such as `unwinder_mark_done`, `get_next_unwinder_after_native_frame`, etc.. Perhaps go with that and call it `unwinder_mark_nonleaf_frame`?